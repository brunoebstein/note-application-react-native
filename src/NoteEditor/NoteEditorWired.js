import { connect } from 'react-redux';

import { selectedNoteSelector } from './selectedNoteSelector';
import { saveNote } from './saveNoteAction';
import { NoteEditor } from './NoteEditor';

const mapStateToProps = state => ({
  note: selectedNoteSelector(state),
});

export const NoteEditorWired = connect(
  mapStateToProps,
  { saveNote },
)(NoteEditor);
