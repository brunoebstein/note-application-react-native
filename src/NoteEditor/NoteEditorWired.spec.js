import { Provider } from 'react-redux';
import { createStore } from 'redux';

import { NoteEditorWired } from './NoteEditorWired';

describe('NoteEditorWired component', () => {
  function matchSnapshotWithProvider(state) {
    const store = createStore(() => state);

    matchSnapshot(
      <Provider store={store}>
        <NoteEditorWired />
      </Provider>,
    );
  }

  it('should render an empty form without selected note', () => {
    const state = {
      notes: [{ title: 'a title', content: 'a content' }],
      selectedNote: null,
    };

    matchSnapshotWithProvider(state);
  });

  it('should render a prefilled form with selected note', () => {
    const state = {
      notes: [{ title: 'a title', content: 'a content' }],
      selectedNote: { title: 'a title', content: 'a content' },
    };

    matchSnapshotWithProvider(state);
  });
});
