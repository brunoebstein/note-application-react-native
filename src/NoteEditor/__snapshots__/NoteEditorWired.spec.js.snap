// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteEditorWired component should render a prefilled form with selected note 1`] = `<Connect(NoteEditor) />`;

exports[`NoteEditorWired component should render an empty form without selected note 1`] = `<Connect(NoteEditor) />`;
